### DISCUSSION TITLE
在 WPF 中 Frame 只有一个能工作，是否有什么方法能够在一个窗口内显示多个子窗口

### DISCUSSION THREAD

**Discussion by XiaoMouz** [BYJRK/DotNet-Discussions#88] [WPF 开发]
我想实现一个类似多窗口的效果，因此设置了三个 Frame，并且使用了4个 StackPanel 将其隔开
```
       <StackPanel Width="1920" Height="790">
            <StackPanel
                Width="1440"
                Height="Auto"
                HorizontalAlignment="Left">
                <Frame Name="mainContainer" NavigationUIVisibility="Hidden" />
            </StackPanel>
            <StackPanel
                Width="480"
                Height="790"
                HorizontalAlignment="Right">
                <StackPanel Height="455" VerticalAlignment="Top">
                    <Frame Name="subContainer" NavigationUIVisibility="Hidden" />
                </StackPanel>
                <StackPanel Height="Auto" VerticalAlignment="Bottom">
                    <TextBlock Text="Bottom" />
                </StackPanel>
            </StackPanel>
        </StackPanel>
```
但是只有 Name 为 `mainContainer` 的 Frame 才能正常显示，在 `mainContainer` 的 Page 发生了异常后 `subContainer` 就显示了...

其实相比起这个我更想知道在 WPF 中如何实现像 WinForm 那种一个程序的 Window 内包含多个窗口的操作，因为我觉得这个操作能从根本上解决掉我的问题（

**Comment by CodingOctocat** [comment]
`Frame` 换成 `Page` 试试。

**Reply by XiaoMouz** [reply]
五一快乐

我尝试修改为 `Page`，但是依旧无法工作，首先的错误原因是 `Page` 的父元素必须为 `Frame` 或 `Window`，而 `Frame` 不允许填充子元素所以换为 `Window`,很显然 `Window` 也不行，因为 `Window` 必须作为根标签存在，并且实际操作后也是如此（

**Reply by CodingOctocat** [reply]
确实，失误了。。。你需要的是 `ContentControl`+`UserControl`、`Frame` 相对重量。

**Comment by BYJRK** [comment]
你的子窗口时是固定的，还是可以拖动的？

如果是固定的，那最简单的方法是添加多个 ContentControl，然后给每一个的 Content 设置不同的 UserControl；
如果不是，那么或许可以考虑使用 AvalonDock 或者 ChildWindow。有挺多 WPF 扩展都提供了类似功能，比如 [Xceed](https://github.com/xceedsoftware/wpftoolkit)

**Reply by XiaoMouz** [reply]
好的，我饭后试试Xceed和别的，是否固定这个需求我不是很清楚所以就当不固定处理比较好

**Reply by XiaoMouz** [reply]
草，竟然是因为布局把 Frame 挤下去了
